javalib/src/main/scala/java/util/ArrayDeque.scala

```diff
@@ -327,7 +326,7 @@ class ArrayDeque[E] private (initialCapacity: Int)
     do {
       if (i >= capacity)
         i = 0
-      if (Objects.equals(inner(i), o))
+      if (Objects.equals(inner(i).asInstanceOf[E], o))
```

Comment:
The cast is probably not necessary here. `Object.equals` accepts any type of argument.